schema-serializer/src/main/java/io/confluent/kafka/serializers/AbstractKafkaSchemaSerDe.java

```diff
@@ -221,10 +241,25 @@ private Stream<? extends RuleBase> initRuleObject(
   }
 
   // Visible for testing
-  public Map<String, RuleExecutor> getRuleExecutors() {
+  public Map<String, Map<String, RuleExecutor>> getRuleExecutors() {
     return ruleExecutors;
   }
 
+  private RuleExecutor getRuleExecutor(RuleContext ctx) {
+    Rule rule = ctx.rule();
+    Map<String, RuleExecutor> executors =
+        ruleExecutors.get(rule.getType().toUpperCase(Locale.ROOT));
+    if (executors != null && !executors.isEmpty()) {
+      RuleExecutor executor = executors.get(rule.getName());
+      if (executor != null) {
+        return executor;
+      } else {
+        return executors.entrySet().iterator().next().getValue();
```

Comment:
Should we throw an exception instead of using the first rule executors?

Reply:
Thanks, added a comment